File: src/java/VulnerableApp.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.Statement;

public class VulnerableApp {

    // 1. Hardcoded credentials (Security Hotspot)
    private static final String PASSWORD = "password123";

    public void getUser(String userId) throws Exception {

        // 2. SQL Injection (Critical Vulnerability)
        Connection conn = DriverManager.getConnection("jdbc:mysql://localhost/db", "root", PASSWORD);
        Statement stmt = conn.createStatement();
        stmt.execute("SELECT * FROM users WHERE id = " + userId);
    }

    public void divide(int a, int b) {
        // 3. Possible division by zero (Bug)
        int result = a / b;
        System.out.println(result);
    }

    public void emptyCatch() {
        try {
            int x = 10 / 0;
        } catch (Exception e) {
            // 4. Empty catch block (Code Smell)
        }
    }
}

File: src/javascript/vulnerable_app.js
const express = require("express");
const app = express();

// 1. Hardcoded API key (Security Hotspot)
const API_KEY = "sk_test_123456";

// 2. SQL Injection (Vulnerability)
app.get("/user", (req, res) => {
  const userId = req.query.id;
  const query = "SELECT * FROM users WHERE id = " + userId;
  res.send(query);
});

// 3. Cross-Site Scripting (Vulnerability)
app.get("/search", (req, res) => {
  res.send("<h1>" + req.query.q + "</h1>");
});

// 4. Weak random generator (Security Hotspot)
function generateToken() {
  return Math.random().toString(36);
}

// 5. Console log in production (Code Smell)
console.log("Server started");

app.listen(3000);